fofpetaio.c

```diff
@@ -77,13 +73,27 @@ void fof_save_particles(int num) {
     MPI_Barrier(MPI_COMM_WORLD);
     fof_write_header(&bf); 
 
+    int stype = -1;
+    int * argind = NULL;
+    int NumPIG = 0;
+
     for(i = 0; i < IOTable.used; i ++) {
         /* only process the particle blocks */
         char blockname[128];
         int ptype = IOTable.ent[i].ptype;
         BigArray array = {0};
         if(ptype < 6 && ptype >= 0) {
+            walltime_measure("/FOF/IO/Write");
             sprintf(blockname, "%d/%s", ptype, IOTable.ent[i].name);
+            if(ptype != stype) {
```

Comment:
Yes it is sorted to avoid rebuilts -- but it's just slightly more twists than necessary. Perhaps add a `FIXME` such that I'll replace it when I have the time to wander the source code -- for the current version already works.